src/state/notesIndexedDBStorageService.js

```diff
@@ -0,0 +1,137 @@
+const DB_NAME = 'notes'
+const DB_VERSION = 1
+const STORE_NAME = 'notes'
+
+let database = null
+
+const notesDataTransformer = {
+  toBuffer (notes) {
+    // input an array of notes objects
+    // output an ArrayBuffer to be written to IDB
+    let notesJSON = ''
+    try {
+      notesJSON = JSON.stringify(notes)
+    } catch (err) {
+      console.error(err)
+      notesJSON = ''
+    }
+    const buffer = new ArrayBuffer(notesJSON.length * 2)
+    const view = new Uint16Array(buffer)
```

Comment:
why 16 bit and not smaller / larger?

Reply:
JavaScript stores strings with 2 bytes per character, thus 16-bit.